frontend/src/components/pages/RequestKit.jsx

```diff
@@ -1,57 +1,55 @@
 import React, { useEffect, useState } from "react";
-import { API_URL, API_URL2 } from "../../constants";
+import { API_URL } from "../../constants";
 import { useLocation, useNavigate } from "react-router-dom";
-import { Link } from "react-router-dom";
 
 function RequestKit({ user }) {
   const location = useLocation();
   const kitId = location.state?.kitId || "";
+  const kitName = location.state?.kitName || "";
 
-  const [orderMessages, setOrderMessages] = useState("");
-
-  // Predefine name, email, and kitId
-  const [firstName, setFirstName] = useState(user ? user.first_name : "");
-  const [lastName, setLastName] = useState(user ? user.last_name : "");
-  const [email, setEmail] = useState(user ? user.email : "");
-  const [, setKitId] = useState(kitId || "");
-
-  // Other inputs
-  const [phone, setPhone] = useState("");
-  const [schoolName, setSchoolName] = useState("");
-  const [schoolAddress, setSchoolAddress] = useState("");
-  const [comments, setComments] = useState("");
-  const [schoolYear, setSchoolYear] = useState("");
+  const [orderMessages, setOrderMessages] = useState("");  
   const ordersUrl = `${API_URL}/orders`;
   const jwt = localStorage.getItem("jwt");
   const navigate = useNavigate();
 
+  const [orderForm, setOrderForm] = useState({
+    phone: "",
+    schoolName: "",
+    schoolAddress: "",
+    comments: "",
+    schoolYear: "",
+    email: user ? user.email : "",
+    firstName: user ? user.first_name : "",
+    lastName: user ? user.last_name : "",
+  });
+
   useEffect(() => {
     window.scrollTo(0, 0);
   }, []);
 
   const handleSubmit = async (e) => {
     e.preventDefault();
 
-    const formData = {
+    
+    const data = {
       order: {
-        phone,
-        school_address: schoolAddress,
-        school_name: schoolName,
-        school_year: schoolYear,
+        phone: orderForm.phone,
+        school_address: orderForm.schoolAddress,
+        school_name: orderForm.schoolName,
+        school_year: orderForm.schoolYear,
         kit_id: kitId,
-        comments,
+        comments: orderForm.comments,
```

Comment:
Not for this PR, but there's an option here to make this code, and any form submission you have even DRY-er, letting you spread your data into the API request. I'll make a ticket for it